Source/Shaders/GlobeFS.glsl

```diff
@@ -191,17 +197,27 @@ vec4 sampleAndBlend(
     alphaMultiplier = step(vec2(0.0), textureCoordinateRectangle.pq - tileTextureCoordinates);
     textureAlpha = textureAlpha * alphaMultiplier.x * alphaMultiplier.y;
 
-#if defined(APPLY_DAY_NIGHT_ALPHA) && defined(ENABLE_DAYNIGHT_SHADING)
-    textureAlpha *= mix(textureDayAlpha, textureNightAlpha, nightBlend);
-#endif
-
     vec2 translation = textureCoordinateTranslationAndScale.xy;
     vec2 scale = textureCoordinateTranslationAndScale.zw;
     vec2 textureCoordinates = tileTextureCoordinates * scale + translation;
     vec4 value = texture2D(textureToSample, textureCoordinates);
     vec3 color = value.rgb;
     float alpha = value.a;
 
+    #ifdef APPLY_MATERIAL
+    if (textureAlpha > 0.0) {
+        // itterate over the layerColor array to check for matching layerIndex and send color value to materialInput
+        //TODO: Find a different way to set pass layerColor values. Here, only the first two layers are handled as this loop leads to performance problems (shaders compilation time)
+        for (int i = 0; i < czm_layerColorNumber; i++) {
+            if (i == layerIndex) {materialInput.layerColor[i] = value; break;}
+        }
+    }
+    #endif
```

Comment:
It's unfortunate that `czm_layerColorNumber` would be capped to the first two layers due to slow compilation.

I can't think of a good way to avoid this in WebGL 1 unless `layerIndex` were somehow made to be a constant expression, and that seems tricky since `GlobeSurfaceShaderSet.prototype.getShaderProgram` doesn't have access to the imagery layers.

Since WebGL2 allows dynamically indexed arrays this would simply be `materialInput.layerColor[layerIndex] = value`. Another reason to switch to WebGL 2 soon.

Reply:
Thanks for the feedback @lilleyse. This pull request is just me thinking out loud about this feature and how best to implement it. In my use case, only one layer is made available to the shader. To work around the limitation of a hard coded number of layers, could this number be set by the user in the form of a "define"?

Reply:
> To work around the limitation of a hard coded number of layers, could this number be set by the user in the form of a "define"?

Possibly. I considered that as well. The only problem is that `materialInput.glsl` is used by other systems that don't have access to the define — though they could mock the define and set it to 1.

I'm not sure about the exact requirements of your application, but would it be acceptable for the material to be a single color after all the imagery layers have been resolved rather than an array of layer colors? That would simplify things quite a bit.

Reply:
My application needs the color of one specific layer, so using the resulting mix of all layers will not be acceptable. In any case, this pull request aims at finding a general solution that can be useful to Cesium users at large. I will investigate a "define" based approach to the problem and post results in the PR.